pkg/services/govmomi/service.go

```diff
@@ -580,3 +765,28 @@ func (vms *VMService) reconcileClusterModuleMembership(ctx *virtualMachineContex
 	}
 	return nil
 }
+
+func markIPAddressClaimedConditionInvalidIPWithError(vm *infrav1.VSphereVM, msg string) (bool, error) {
+	conditions.MarkFalse(vm,
+		infrav1.IPAddressClaimedCondition,
+		infrav1.IPAddressInvalidReason,
+		clusterv1.ConditionSeverityError,
+		msg)
+	return false, errors.New(msg)
+}
+
+func markIPAddressClaimedConditionWaitingForClaimAddress(vm *infrav1.VSphereVM, msg string) {
+	conditions.MarkFalse(vm,
+		infrav1.IPAddressClaimedCondition,
+		infrav1.WaitingForIPAddressReason,
+		clusterv1.ConditionSeverityInfo,
+		msg)
+}
+
+func isIPv4(ip net.IP) bool {
+	return ip.To4() != nil
```

Comment:
FYI: There is a new `netip` package since go1.18 with a nicer API: https://pkg.go.dev/net/netip#Addr.Is4

Reply:
cool, we've switched over to `netip`